### ISSUE TITLE
F-Droid reproducible build failed

### ISSUE THREAD

**Issue by linsui** [elastic-rock/KeepScreenOn#26]
https://gitlab.com/fdroid/fdroiddata/-/issues/3267 We don't know how to fix it currently. It seems it's caused by toolchain update and the reproducible-apk-tools doesn't work with that yet. Could you please take a look? Thanks!

**Comment by elastic-rock** [comment]
Thank you for opening this issue. Unfortunately, my knowledge in this is rather limited so I am affraid I won't be able to help much.

**Comment by licaon-kter** [comment]
Trying all sorts of commands combos

but...

Where are these files from anyway? What are they?
```
fixing 'META-INF/services/P1.w'...
fixing 'META-INF/services/Q1.a'...
```

**Comment by linsui** [comment]
They are java spi minigied by r8.

**Comment by elastic-rock** [comment]
Thank you very much. Once there is a known cause for the issue, let me know.

**Comment by elastic-rock** [comment]
One thing I notice is that I get this gradle warning:

`SDK processing. This version only understands SDK XML versions up to 3 but an SDK XML file of version 4 was encountered. This can happen if you use versions of Android Studio and the command-line tools that were released at different times.`

**Comment by linsui** [comment]
It turns out that the new AGP causes this problem. https://github.com/obfusk/reproducible-apk-tools/issues/19

**Comment by licaon-kter** [comment]
please reopen

ref: https://monitor.f-droid.org/builds/log/com.elasticrock.keepscreenon/29#site-footer

diff log: [elas29.log](https://github.com/user-attachments/files/16548261/elas29.log)

**Comment by linsui** [comment]
1.19.0 also failed. 
[com.elasticrock.keepscreenon_v1.19.0.apk.html.zip](https://github.com/user-attachments/files/16825042/com.elasticrock.keepscreenon_v1.19.0.apk.html.zip)

Maybe it's caused by cache. Could you please clean the cache and rebuild the apk?

**Comment by licaon-kter** [comment]
@elastic-rock fun thought, can you build 1.19 with `false` here https://github.com/elastic-rock/KeepScreenOn/blob/v1.19.0/app/build.gradle.kts#L29 and attach the APK for us to test?

**Comment by elastic-rock** [comment]
> @elastic-rock fun thought, can you build 1.19 with `false` here https://github.com/elastic-rock/KeepScreenOn/blob/v1.19.0/app/build.gradle.kts#L29 and attach the APK for us to test?

Here it is. Thanks! (note that I also had to set isShrinkResources to false)
[keepscreenon_v1.19.0_minifyfalse.zip](https://github.com/user-attachments/files/16834705/keepscreenon_v1.19.0_minifyfalse.zip)

**Comment by licaon-kter** [comment]
@elastic-rock indeed, as with other recent apps, minify introduces non-determinism :disappointed: 

the attached APK is repro, then again, it's 10 times bigger too

**Comment by linsui** [comment]
The `res/Hz` file is not reproducible. I guess it's generated by `com.google.android.gms:oss-licenses-plugin`.

**Comment by elastic-rock** [comment]
Do you have any proposals as to what could be done with this whole situation? 🙂 

I disabled `minify` but this was only meant as a temporary solution since an APK size of 10 MB and install size of 131 MB is just not feasible in the long term. Well and now I hindered my hopes for a finaly successful build with the `oss-licenses-plugin.` 🙃 I suppose I have to figure out a diferent solution for the licenses and regarding `minify` , are there any fixes coming?

Thanks!

**Comment by elastic-rock** [comment]
I switched the `oss-licenses-plugin` for `com.mikepenz.aboutlibraries`. If the issue with minify still persists, let me know and I can try and build the app outside Android Studio using gradle directly.

**Comment by licaon-kter** [comment]
@elastic-rock do add https://github.com/mikepenz/AboutLibraries/issues/784#issuecomment-1205501876 else the files have compile dates in them

**Comment by elastic-rock** [comment]
Done. Thanks! :slightly_smiling_face:

**Comment by elastic-rock** [comment]
Fix added in `1.20.3`

**Comment by elastic-rock** [comment]
Amazing! Thanks for the help, you have my respect! 😃
